### PR TITLE
[ISSUE #15761] Reconcile historical MCP lifecycle resources

### PR DESCRIPTION
**Pull request by KomachiSion**
## What is the purpose of the change

For #15761.

Continue #15761 by hosting historical MCP management metadata in the shared AI Resource lifecycle tables without changing the existing serving plane.

This is the historical reconciliation stage after #15774. It maps legacy MCP manifests and their existing Server/Tools/Resources Config coordinates into `ai_resource` and `ai_resource_version`, while management routes remain in `SYNCING`. No HTTP, gRPC, SDK, gateway, payload Config, or Naming contract changes in this pull request.

## Brief changelog

* add a cluster-singleton, renewable-lease reconciliation task started after the root application is ready;
* page all namespaces and serving manifests through `McpServingManifestStorage`;
* validate the historical manifest, Server, Tools, and Resources content before creating zero-copy storage descriptors;
* reconcile Versions first and Resource metadata last, with idempotent retry, CAS recovery, duplicate/conflict detection, and orphan diagnostics;
* persist non-authoritative `SYNCING` progress, including zero-difference, pending Search backfill, and a permanent `managedCutoverReady=false` guard;
* never rewrite historical payload bytes, mutate Naming, delete orphan rows, enqueue the old ID-keyed Search projection, or write the `LIFECYCLE_MANAGED` marker;
* update the English and Chinese MCP specs for the lease/progress contract and staged reconciliation boundary;
* add unit tests for the reconciler and lease/progress state machine, and close the missing/partial coverage reported by Codecov on #15774.

## Verifying this change

* `mvn -pl ai spotless:apply spotless:check`
* focused MCP/storage/reconciliation suite: 144 tests passed;
* repository pre-submission gate: all 62 modules passed `clean compile apache-rat:check checkstyle:check spotbugs:check spotless:check -DskipTests`;
* focused JaCoCo results:
  * prior #15774 mapping/storage classes: 100% line coverage;
  * `McpHistoricalResourceReconciler`: 100% line, 78.5% branch coverage;
  * `McpLifecycleReconciliationTask`: 99.5% line, 80.6% branch coverage.

The full `ai` suite ran 2,376 tests. All changed tests passed; the pre-existing order-sensitive `OpenAiCompatibleResourceIndexEnhancementServiceTest` reported one failure and one error in the full-suite order, while its isolated rerun passed 7/7. The changed code does not access that test's configuration or system properties.

Follow this checklist to help us incorporate your contribution quickly and easily:

* [x] A GitHub issue is filed for the change.
* [x] The pull request title follows the issue title convention.
* [x] The description explains the behavior and compatibility boundary.
* [x] Unit tests cover the new behavior and previous coverage gaps.
* [x] Spotless and the repository pre-submission checks pass.

### CODE REVIEW

File: ai/src/main/java/com/alibaba/nacos/ai/config/McpLifecycleReconciliationTask.java [hunk 1]
```diff
@@ -0,0 +1,565 @@
...
+            publishLease(content, true, current.md5);
+            LOGGER.info("Took over expired MCP lifecycle reconciliation lease");
+            return new ReconciliationLease(owner, contentMd5(content));
+        } catch (Exception e) {
+            LOGGER.debug("MCP lifecycle reconciliation lease was taken by another node");
+            return null;
+        }
+    }
+    
+    private LeaseRecord readLease() {
```
Comment (nacos-community) [inline ai/src/main/java/com/alibaba/nacos/ai/config/McpLifecycleReconciliationTask.java:363]: **[Warning]** If the lease config content ever becomes unparseable, `LeaseRecord.parse` throws and `readLease` swallows it into `null` (line ~368). `tryAcquireLease` then takes the `current == null` branch, `publishLease(content, false, null)` fails with `ConfigAlreadyExistsException` (verified against `ConfigOperationService.publishConfig` → `addConfigInfo` path), and the task silently returns `null` every cycle — reconciliation is halted indefinitely with only a DEBUG-level "another node owns the lease" message.

Suggestion: treat unparseable (but present) lease content as takeover-eligible via the CAS path, or at minimum log a WARN so operators can detect the stuck state.

File: ai/src/main/java/com/alibaba/nacos/ai/service/mcp/storage/McpServingManifestStorage.java [hunk 2]
```diff
@@ -154,6 +206,28 @@ private ConfigFormV3 buildForm(String namespaceId, McpServerVersionInfo manifest
...
+        throws NacosException {
+        if (configInfo == null) {
+            throw storageFailure("MCP serving Manifest page contains an empty row", null);
+        }
+        final McpServerVersionInfo manifest;
+        try {
+            manifest = JacksonUtils.toObj(configInfo.getContent(), McpServerVersionInfo.class);
+            validateManifest(manifest);
+        } catch (NacosDeserializationException | IllegalArgumentException e) {
+            throw storageFailure("MCP serving Manifest page contains invalid content", e);
```
Comment (nacos-community) [inline ai/src/main/java/com/alibaba/nacos/ai/service/mcp/storage/McpServingManifestStorage.java:234]: **[Warning]** A single corrupted row in the manifest page aborts the whole scan: this `storageFailure` propagates through `reconcileNamespace` (the `list()` call is outside its per-manifest try/catch) into `executeReconciliation`'s outer catch, which skips **all remaining namespaces** for this generation.

That bypasses the per-manifest fault isolation the reconciler otherwise provides (`reconcileNamespace` catches `reconciler.reconcile` failures per manifest and records diagnostics). For a task whose purpose is cleaning up messy historical state, one bad legacy config would block reconciliation everywhere until manually fixed.

Suggestion: catch per-row decode failures inside `list()` (or in the task loop) and surface them as diagnostics instead of aborting the generation.

File: ai/src/main/java/com/alibaba/nacos/ai/config/McpLifecycleReconciliationTask.java [hunk 3]
```diff
@@ -0,0 +1,565 @@
...
+            if (!owned.get()) {
+                return;
+            }
+            try {
+                String content = leaseContent(owner);
+                publishLease(content, true, currentMd5);
+                currentMd5 = contentMd5(content);
+            } catch (Exception e) {
+                owned.set(false);
+                LOGGER.warn("Failed to renew MCP lifecycle reconciliation lease", e);
```
Comment (nacos-community) [inline ai/src/main/java/com/alibaba/nacos/ai/config/McpLifecycleReconciliationTask.java:537]: **[Info]** `renewSafely` drops the lease (`owned=false`) on the first transient publish failure, aborting the whole generation, and this node cannot re-acquire until the 10-minute TTL expires (the lease still names it as owner). This is conservatively safe, so just a consideration: a bounded retry within the remaining TTL would tolerate brief DB/network hiccups without sacrificing the split-brain protection the CAS md5 already provides.

**Review by nacos-community — COMMENTED** [review]
### Summary
Adds a cluster-singleton, renewable-lease reconciliation task that maps historical MCP serving manifests into `ai_resource`/`ai_resource_version` rows while staying in `SYNCING` (zero payload rewrite, no Naming/route changes). The design is defensive — CAS lease takeover, idempotent insert-with-recovery, version-rows-first/resource-last ordering, duplicate/orphan diagnostics — and it ships with extensive tests plus matching `specs/` updates. Overall high quality; two non-blocking edge-case robustness gaps found, details inline.

### Findings
- **[Warning]** `McpLifecycleReconciliationTask.java:349` — unparseable lease content silently halts reconciliation forever (see inline)
- **[Warning]** `McpServingManifestStorage.java:219` — one corrupted manifest row aborts reconciliation for all remaining namespaces in the generation (see inline)
- **[Info]** `McpLifecycleReconciliationTask.java:523` — single transient lease-renew failure drops the lease for the rest of the TTL; bounded retry worth considering

### Verification notes
- Checked against `develop@39d94fd`: `ConfigOperationService.publishConfig` semantics match the task's usage (create-only via `addConfigInfo` → `ConfigAlreadyExistsException`; CAS renew/takeover via `insertOrUpdateCas`; progress publish is a plain upsert, so first-run persistence works).
- `AiResourcePersistService`/`AiResourceVersionPersistService` signatures (`insert`, `updateMetaCas`) match the reconciler call sites; `metaVersion` null is guarded before unboxing.
- The `McpJsonValidationSupport.parseSingleValue` relaxation (blank check → parser-only) is behavior-equivalent: blank input is still rejected by `validateSingleJsonValue`.
- Spec updates (`specs/en|zh-cn/ai/mcp-server-spec.md`) accompany the lease/progress contract change, per repo convention.

### Suggestions
1. Make corrupted lease content detectable or takeover-eligible (finding 1).
2. Demote per-row manifest decode failures to diagnostics so one bad legacy config cannot block other namespaces (finding 2).

---
*Automated review by github-manager-bot*
